pycbc/types/timeseries.py

```diff
@@ -478,25 +478,56 @@ def psd(self, segment_duration, **kwds):
                            seg_stride=seg_stride,
                            **kwds)
 
-    def gate(self, time, zero_width=0.25, taper_width=0.25):
+    def gate(self, time, window=0.25, method='taper', inplace=False,
+             taper_width=0.25, invpsd=None):
         """ Gate out portion of time series
 
         Parameters
         ----------
         time: float
             Central time of the gate in seconds
-        zero_width: float
-            Half-length in seconds of zeros around gate.
+        window: float
+            Half-length in seconds to remove data around gate time.
+        method: str
+            Method to apply gate, options are 'hard', 'taper', and 'paint'.
+        inplace: bool
+            If True, do operations inplace to this time series, else return
+            new time series.
         taper_width: float
-            Lenght of tapering region on either side of zero'd data
+            Length of tapering region on either side of excized data. Only
+            applies to the taper gating method.
+        invpsd: pycbc.types.FrequencySeries
+            The inverse PSD to use for painting method. If not given,
+            a PSD is generated using default settings.
 
         Returns
         -------
         data: pycbc.types.TimeSeris
             Gated time series
         """
-        from pycbc.strain import gate_data
-        return gate_data(self.copy(), [(time, zero_width, taper_width)])
+        data = self if inplace else self.copy()
+        if method == 'taper':
+            from pycbc.strain import gate_data
+            return gate_data(data, [(time, window, taper_width)])
+        elif method == 'paint':
+            # Uses the hole-filling method of
+            # https://arxiv.org/pdf/1908.05644.pdf
+            from pycbc.strain.gate import gate_and_paint
+            if invpsd is None:
+                # These are some bare minimum settings, normally you
+                # should probably provide a psd
+                invpsd = 1. / self.filter_psd(self.duration/32, self.delta_f, 0)
+            lindex = int((time - window - self.start_time) / self.delta_t)
+            rindex = lindex + int(2 * window / self.delta_t)
+            lindex = lindex if lindex >=0 else 0
+            rindex = rindex if rindex <= len(self) else len(self)
+            return gate_and_paint(data, lindex, rindex, invpsd, copy=False)
```

Comment:
This should be `copy=not inplace`.

Reply:
Done

Reply:
Huh? `copy` is still hard-coded to `False` here.

Reply:
My original comment was referring to the fact that the copy keyword argument in the call to `gate_and_paint` wasn't respecting the `inplace` argument, not that you needed to change the name of `inplace` to `copy`.

Reply:
Ok, then I misunderstood, and so did you. This is correct. The copy is done at the top for all the types. @cdcapano  Look at line 508

Reply:
Oh! Sorry, I missed that. Looks fine then.